service/src/main/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/EncounterMapper.java

```diff
@@ -3,17 +3,239 @@
 import static org.hl7.fhir.dstu3.model.Encounter.EncounterStatus.FINISHED;
 import static org.hl7.fhir.dstu3.model.IdType.newRandomUuid;
 
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import uk.nhs.connect.iucds.cda.ucr.IVLTS;
+import uk.nhs.connect.iucds.cda.ucr.POCDMT000002UK01EncompassingEncounter;
+
 import org.hl7.fhir.dstu3.model.Encounter;
+import org.hl7.fhir.dstu3.model.Period;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 
 @Component
 public class EncounterMapper {
 
-    public Encounter mapEncounter() {
+    @Autowired
+    PeriodMapper periodMapper;
+
+    @Autowired
+    ParticipantMapper participantMapper;
```

Comment:
See Java coding standards. Let's try to use constructors instead of field injection.
Field should be private